snapcraft/storeapi/assertions.py

```diff
@@ -0,0 +1,262 @@
+            'until': '2018-02-10T08:35:00.390258Z'
+           }],
+        }
+
+    """
+
+    _assertion_type = 'developers'
+
+    def new_assertion(self, *, developers):
+        """Create a new assertion with developers based out of the current one.
+
+        The new assertion has its assertion's authority-id normalized to the
+        assertion's publisher_id and the assertion's revision increaded by 1.
+
+        :param list developers: a list of a dictionary developers holding the
+                                keys: developer_id (mandatory), since
+                                (mandatory) and until (optional).
+        :returns: a new assertion based out of the current assertion with the
+                  provided developers list.
+        :rtype: DeveloperAssertion
+        """
+        new_assertion = deepcopy(self.assertion)
+        new_assertion['developers'] = developers
+
+        # The revision should be incremented, to avoid `invalid-revision`
+        # errors.
+        new_assertion['revision'] = str(int(
+            self.assertion.get('revision', '0'))+1)
+
+        # There is a possibility that the `authority-id` to be `canonical`,
+        # which should be changed to the `publisher_id` to match the signing
+        # key.
+        new_assertion['authority-id'] = self.publisher_id
+
+        new_instance = DeveloperAssertion(snap_name=self._snap_name,
+                                          signing_key=self._signing_key)
+        # Reference the already fetched information
+        new_instance._account_info = self._account_info
+        new_instance.assertion = new_assertion
+        return new_instance
+
+    def get(self):
+        """Returns a dict containing the developer assertion for snap_name.
+
+        The data that comes from the store query looks as follows:
+        {'snap_developer': {
+             'type': 'snap-developer',
+             'authority-id': <account_id of the publisher>,
+             'publisher-id': <account_id of the publisher>,
+             'snap-id': 'snap_id',
+             'developers': [{
+                 'developer-id': 'account_id of dev-1',
+                 'since': '2017-02-10T08:35:00.390258Z'
+                },{
+                 'developer-id': 'account_id of dev-2',
+                 'since': '2017-02-10T08:35:00.390258Z',
+                 'until': '2018-02-10T08:35:00.390258Z'
+                }],
+             }
+        }
+
+        The assertion is saved without the snap_developer header.
+
+        :returns: the latest developer assertion corresponding to snap_name.
+        :rtype: dict
+        """
+        try:
+            self.assertion = super().get()
+        except errors.StoreValidationError as e:
+            if e.error_list[0]['code'] != 'snap-developer-not-found':
+                raise
+            self.assertion = {
+                'type': 'snap-developer',
+                'authority-id': self.publisher_id,
```

Comment:
w.r.t. comment above about the "canonical" authority, and just to be super clear ... This bit is correct. If the assertion is generated locally then it should be whoever authenticated with snapcraft.